src/sentry/tasks/assemble.py

```diff
@@ -474,7 +474,17 @@ def _create_artifact_bundle(self) -> None:
 
         # We want to measure how much time it takes to extract debug ids from manifest.
         with metrics.timer("tasks.assemble.artifact_bundle.extract_debug_ids"):
-            bundle_id, debug_ids_with_types = self.archive.extract_debug_ids_from_manifest()
+            debug_ids_with_types = self.archive.extract_debug_ids_from_manifest()
+
+        bundle_id = self.archive.extract_bundle_id()
+        if not bundle_id:
+            # In case we didn't find the bundle_id in the manifest, we will just generate our own.
+            # XXX: can this ever fail / return `None`?
+            bundle_id = ArtifactBundleArchive.normalize_debug_id(
+                self.assemble_result.bundle.checksum
+            )
```

Comment:
This fallback is great for de-duplication, didn't think about using it, nice!

The only concern I have is the format, a checksum isn't a UUID from my knowledge of the system, thus it wouldn't make sense to normalize it. Am I missing something?

Reply:
The normalization works fine, as a debug-id has an additional appendix. However we need to strip *that* away for it to be a valid uuid that we can save into the database.